lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/REEFEnvironment.java

```diff
@@ -86,11 +90,12 @@ public static REEFEnvironment fromConfiguration(final Configuration... configura
     injector.getInstance(REEFVersion.class).logVersion();
 
     final REEFErrorHandler errorHandler = injector.getInstance(REEFErrorHandler.class);
+    final JobStatusHandler jobStatusHandler = injector.getInstance(JobStatusHandler.class);
 
     try {
 
       final Clock clock = injector.getInstance(Clock.class);
-      return new REEFEnvironment(clock, errorHandler);
+      return new REEFEnvironment(clock, errorHandler, jobStatusHandler);
```

Comment:
What if a user wants to bind their own `JobStatusHandler` and, for that matter `REEFErrorHandler`? Maybe it makes sense to provide overloads for the factory methods?